conans/client/generators/cmake_common.py

```diff
@@ -826,19 +826,24 @@ class CMakeCommonMacros:
     """)
 
     apple_frameworks_macro = textwrap.dedent("""
-        macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS SUFFIX)
+        macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS SUFFIX BUILD_TYPE)
             if(APPLE)
                 if(CMAKE_BUILD_TYPE)
-                    if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
+                    set(_BTYPE ${CMAKE_BUILD_TYPE})
+                elseif(NOT BUILD_TYPE STREQUAL "")
+                    set(_BTYPE ${BUILD_TYPE})
+                endif()
+                if(_BTYPE)
+                    if(${_BTYPE} MATCHES "Debug|_DEBUG")
                         set(CONAN_FRAMEWORKS${SUFFIX} ${CONAN_FRAMEWORKS${SUFFIX}_DEBUG} ${CONAN_FRAMEWORKS${SUFFIX}})
                         set(CONAN_FRAMEWORK_DIRS${SUFFIX} ${CONAN_FRAMEWORK_DIRS${SUFFIX}_DEBUG} ${CONAN_FRAMEWORK_DIRS${SUFFIX}})
-                    elseif(${CMAKE_BUILD_TYPE} MATCHES "Release")
+                    elseif(${_BTYPE} MATCHES "Release|_RELEASE")
```

Comment:
Yes, this needs to be refactored, improved at some point, but I understand that now it is better to move forward.